android/src/main/java/com/appboy/reactbridge/AppboyReactBridge.java

```diff
@@ -599,6 +616,174 @@ public void getInstallTrackingId(Callback callback) {
     reportResultWithCallback(callback, null, Appboy.getInstance(getReactApplicationContext()).getInstallTrackingId());
   }
 
+  @ReactMethod
+  public void getContentCards(final Promise promise) {
+    final IEventSubscriber<ContentCardsUpdatedEvent> subscriber = new IEventSubscriber<ContentCardsUpdatedEvent>() {
+      @Override
+      public void trigger(ContentCardsUpdatedEvent event) {
+        promise.resolve(mapContentCards(event.getAllCards()));
```

Comment:
Shouldn't the `promise.resolve` be placed as the last method call?

Reply:
I tested this and I think we should be ok. first getting the cards and *if neede* update them and then remove the subsciption. do you see any potential issue here?